src/mv_memory.rs

```diff
@@ -184,7 +185,7 @@ impl MvMemory {
         }
     }
 
-    pub(crate) fn consume_lazy_addresses(&self) -> impl IntoIterator<Item = Address> {
-        std::mem::take(&mut *self.lazy_addresses.lock().unwrap()).into_iter()
+    pub(crate) fn consume_lazy_locations(&self) -> impl IntoIterator<Item = MemoryLocation> {
+        std::mem::take(&mut *self.lazy_locations.lock().unwrap()).into_iter()
```

Comment:
It may be faster to just return references here, since:
- We're cloning in post-processing anyway.
- This will be defer-dropped so pre-cleaning it may add latency instead of saving it.